### PR TITLE
factor out HUD widget generation from HU_Drawer() into HU_Ticker()

### LINKED ISSUE #882: factor out HUD widget generation from HU_Drawer() into HU_Ticker()

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #882

### CODE REVIEW

File: src/hu_stuff.c [hunk 1]
```diff
@@ -885,6 +885,449 @@ static int HU_top(int i, int idx1, int top1)
   return i;
 }
 
+// do the hud ammo display
+static void HU_widget_build_ammo (void)
+{
+  char ammostr[80], *s; //jff 3/8/98 allow plenty room for dehacked mods
```
Comment (fabiangreffrath) [inline src/hu_stuff.c:891]: Does anybody know why we need this `ammostr[80]` *and* `hud_ammostr[80]`? And what does this have to do with dehacked?

File: src/hu_stuff.c [hunk 2]
```diff
@@ -885,6 +885,449 @@ static int HU_top(int i, int idx1, int top1)
...
+      // put it in the init string
+      i = HU_top(i, idx3, top3);
+
+      // if the fourth biggest number exists,
+      // put it in the init string
+      i = HU_top(i, idx4, top4);
+
+      hud_keysstr[i] = '\0';
+    } //jff 3/17/98 end of deathmatch clause
+    else // build alphabetical key display (not used currently)
```
Comment (fabiangreffrath) [inline src/hu_stuff.c:1257]: This is currently dead code, but sounds interesting. Maybe we should revive it with a config flag?

Comment (rfomin) [reply src/hu_stuff.c:1257]: I think this is an opportunity to delete it 😄

Overall like the changes, I haven’t found any bugs yet.

Comment (fabiangreffrath) [reply src/hu_stuff.c:1257]: > I think this is an opportunity to delete it 😄

Yep, even better!

> Overall like the changes, I haven’t found any bugs yet.

Excellent! See how handy `HU_Drawer()` has become?

Comment (rfomin) [reply src/hu_stuff.c:1257]: > See how handy `HU_Drawer()` has become?

Yes, much better 👍 We can remove this comment now: `// [FG] ~440 lines below`

File: src/hu_stuff.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/hu_stuff.c:889]: This is even still present in PrBoom+. But why should we carry two equally sized string buffer?

Comment (rfomin) [reply src/hu_stuff.c]: Theoretically, it is possible to significantly increase the quantity/max. ammo in DEHACKED, maybe that's why there are these comments. What if we increase size of `hud_ammostr`?

**Review by rfomin — APPROVED** [review]
LGTM! But I must admit that I did not test it for long.

File: src/hu_stuff.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/hu_stuff.c]: > What if we increase size of `hud_ammostr`?

To what, even more than 80 bytes?

Comment (rfomin) [reply src/hu_stuff.c]: Oh yes, with an 8x8 font size, there is no point in increasing the buffer size.
